client/asset/interface.go

```diff
@@ -227,6 +263,23 @@ type Wallet interface {
 	RegFeeConfirmations(ctx context.Context, coinID dex.Bytes) (confs uint32, err error)
 }
 
+// Rescanner is a wallet implementation with rescan functionality.
+type Rescanner interface {
+	Rescan(ctx context.Context) error
+}
+
+// Sender is a wallet that can send funds to an address, as opposed to
```

Comment:
I don't understand this comment.. isn't sending funds to an address the same as withdrawing?

Reply:
The difference is where fees come from.  Withdraw means the recipient gets less than the specified amount, but the dex wallet is reduced by exactly that amount).  Send means the recipient gets exactly the specified amount, but the dex wallet is reduced by a larger amount, determined after mining.  Which one to use depends entirely on the user's intended outcome.  Many wallet's support both.

A UI I imagine would be just a single Withdraw button, but the dialog/form for this would have a checkbox to indicate if the fees should be subtracted from the amount sent to the destination address or a larger amount should be consumed from the wallet to ensure the destination gets the exact amount.  If a wallet only supported Withdraw, it would be a disabled unchecked checkbox.  If it supported only Send, it would be a grayed/unchangable checked checkbox.  If it supported both, it would be user selectable.  To do this means there would have to be another trait to indicate if the wallet can do withdraw with fees subtracted as with BTC and DCR's

Reply:
I see, that makes sense.